scipy/stats/_continuous_distns.py

```diff
@@ -6121,15 +6122,88 @@ def _entropy(self, c):
     def fit(self, data, *args, **kwds):
         parameters = _check_fit_input_parameters(self, data, args, kwds)
         data, fshape, floc, fscale = parameters
-        if floc is None:
+
+        if floc is None and fscale is None:
+            # Optimization over a system of equations is possible when the
+            # location of the data is not set. The optimization requires that
+            # the scale be free, but the shape may be fixed. The equations
+            # are from the partial derivatives of the log-likelihood function.
+
+            def get_shape(scale, location):
+                # The partial derivative with respect to the shape can be
+                # solved in closed form for the shape.
+                if fshape is None:
+                    return (data.shape[0] /
+                            np.sum(np.log((data - location) / scale)))
+                else:
+                    # if `fshape` is fixed, return that instead.
+                    return fshape
+
+            def get_location(scale):
+                # Return the numerically optimal location.
+                return np.min(data) - scale
```

Comment:
```suggestion
            def get_location(scale):
                # Return the numerically optimal location.
                return np.min(data) - scale
```
Considering this calculation is needed below (see [here](https://github.com/mdhaber/scipy/pull/68/files#r792360967)), we could move this out of the `if` statement and use it in both places. (On the other hand, it's so simple you might just write it out in the two places it's needed. Up to you.)

Reply:
Implemented